File: easy/1000-1499/1290-convert-binary-number-in-linked-list-to-integer/sol1.cpp
#include <bits/stdc++.h>
using namespace std;

struct ListNode
{
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};

class Solution
{
public:
    int getDecimalValue(ListNode *head)
    {
        int sum = 0;
        while (head)
        {
            sum *= 2;
            sum += head->val;
            head = head->next;
        }

        return sum;
    }
};

int main()
{
    cout << "Hello C++" << endl;
    return 0;
}

File: medium/2000-2499/2130-maximum-twin-sum-of-a-linked-list/sol1.py
from typing import Optional

# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

# O(n) time complexity, O(n) space complexity
class Solution:
    def pairSum(self, head: Optional[ListNode]) -> int:
        arr = []
        curr = head
        while curr:
            arr.append(curr.val)
            curr = curr.next
        ans = -1
        n = len(arr)
        for i in range(n // 2):
            ans = max(ans, arr[i] + arr[n - 1 - i])
        return ans